cloud-function/src/handlers/proxy-content.ts

```diff
@@ -10,9 +10,10 @@ import { Source, sourceUri } from "../env.js";
 import { PROXY_TIMEOUT } from "../constants.js";
 import { isLiveSampleURL } from "../utils.js";
 
-const NOT_FOUND_PATH = "en-us/_spas/404.html";
+const NOT_FOUND_JSON = "en-us/_spas/404.json";
+const NOT_FOUND_HTML = "en-us/_spas/404.html";
 
-let notFoundBuffer: ArrayBuffer;
+const bufferCache = new Map<string, ArrayBuffer>();
```

Comment:
This prepares for having per-locale 404 pages, as currently the 404 page only has `en-US` links in the top navigation.